app/controllers/FileFailedChecksController.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import pages.{ConversationIdPage, ValidXMLPage}
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import viewmodels.FileStatusViewModel
+import views.html.{FileFailedChecksView, ThereIsAProblemView}
+
+import javax.inject.Inject
+
+class FileFailedChecksController @Inject() (
+  override val messagesApi: MessagesApi,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  val controllerComponents: MessagesControllerComponents,
+  view: FileFailedChecksView,
+  errorView: ThereIsAProblemView
+) extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData() andThen requireData) {
+    implicit request =>
+      (request.userAnswers.get(ValidXMLPage), request.userAnswers.get(ConversationIdPage)) match {
+        case (Some(xmlDetails), Some(conversationId)) =>
+          val action  = routes.FileRejectedController.onPageLoad(conversationId).url
+          val summary = FileStatusViewModel.createFileSummary(xmlDetails.fileName, "Rejected")
```

Comment:
@soljeggers Instead of placing it createFileSummary in FileStatusViewModel can we have a separate FileCheckViewModel